src/devenv/cli.py

```diff
@@ -0,0 +1,445 @@
+import os
+import shutil
+import subprocess
+import time
+import re
+from pathlib import Path
+import pkgutil
+import json
+
+import click
+import terminaltables
+
+
+from .yaml import validate_and_parse_yaml
+
+
+NIX_FLAGS = [
+    "--show-trace",
+    "--extra-experimental-features",
+    "\"nix-command flakes\"",
+    "--option",
+    "warn-dirty",
+    "false",
+]
+FILE = pkgutil.get_loader(__package__).load_module(__package__).__file__
+if 'site-packages' in FILE:
+    SRC_DIR = Path(FILE, '..', '..', 'src')
+else:
+    SRC_DIR = Path(FILE, '..', '..')
+MODULES_DIR = (SRC_DIR / 'modules').resolve()
+FLAKE_FILE_TEMPL = os.path.join(MODULES_DIR, "flake.tmpl.nix")
+FLAKE_FILE = ".devenv.flake.nix"
+FLAKE_LOCK = "devenv.lock"
+
+# define system like x86_64-linux
+SYSTEM = os.uname().machine.lower().replace("arm", "aarch") + "-" + os.uname().sysname.lower()
+
+def run_nix(command: str) -> str:
+    ctx = click.get_current_context()
+    nix_flags = ctx.obj['nix_flags']
+    flags = " ".join(NIX_FLAGS) + " " + " ".join(nix_flags)
+    command_flags = " ".join(ctx.obj['command_flags'])
+    return run_command(f"nix {flags} {command} {command_flags}")
+
+def run_command(command: str) -> str:
+    if command.startswith("nix"):
+        if os.environ.get("DEVENV_NIX"):
+            nix = os.path.join(os.environ["DEVENV_NIX"], "bin")
+            command = f"{nix}/{command}"
+        else:
+            click.echo("$DEVENV_NIX is not set, but required as devenv doesn't work without a few Nix patches.", err=True)
+            click.echo("Please follow https://devenv.sh/getting-started/ to install devenv.", err=True)
+            exit(1)
+    try:
+        return subprocess.run(
+            command, 
+            shell=True,
+            check=True, 
+            env=os.environ.copy(),
+            stdout=subprocess.PIPE,
```

Comment:
We should pipe everything so also stderr and stdin. So I can cancel the nix process with STRG + C 😅